### PR TITLE
Reorder steps on TC-SU-2.7 to use one single OTA image

### LINKED ISSUE #799: TC-SU-2.7: Re-arrange for single image
See discussion here: https://github.com/project-chip/connectedhomeip/pull/71688#discussion_r3762225981

It would be good if the entire suite of software update tests both only applied a single update (for time reasons) and only required the testers to prepare a single update (for ease of test prep). Investigate if it is possible to move the image transfer to the end of the test so the tests can be ordered in a way that only requires a single image.

### PR DESCRIPTION
**Pull request by jtrejoespinoza-grid**
### Summary

Reordered the test steps in `TC_SU_2_7.py` so the full end-to-end update/`VersionApplied` verification runs last, and simplified the test to use a single OTA image instead of two.

Before:
  -   The full update-and-apply verification (Querying → Downloading → Applying → Idle, plus the `VersionApplied` event check) was **step 1**, run before all the edge-case steps (busy query, query timeout, user consent, download error, delayed apply).
  -   This ran the DUT through a full image application at the very start of the test, then exercised failure/edge-case flows against a device already on the updated version.
  -   The test also required **two distinct OTA images** (`ota_image` and `ota_image_extra`) so the full-update step (step 1) and the busy-query step (old step 2) could each apply a different version. This doubled the PIXIT/CI-arg surface (`ota_image_extra`, `expected_software_version_extra`) for no functional benefit to the edge-case steps, which don't depend on which image is served.

Now:

- Edge cases done first without updating the device.
- Full update done in the last step
- Device reboots between steps to clear Delayed states.
- Remove extra ota image, now to needed.

### Testing

Test script must complete:

```
python3 scripts/tests/run_python_test.py --load-from-env /tmp/test_env.yaml --script src/python_testing/TC_SU_2_7.py
```

Reordered steps:

Screenshot 2026-08-18 at 4 22 36 p m

### CODE REVIEW

File: src/python_testing/TC_SU_2_7.py [hunk 1]
```diff
@@ -136,19 +129,17 @@ async def setup_test(self):
...
             provider_app_path=self.provider_app_path,
             ota_image_path=self.ota_image,
             setup_pincode=self.provider_setup_pincode,
             discriminator=self.provider_discriminator,
-            port=self.provider_port,
+            port=self.provider_port, extra_args=['--delayedQueryActionTimeSec', '10', '--queryImageStatus', 'busy'],
             kvs_path=self.provider_kvs_path,
             log_file=self.provider_log,
-            timeout=10
+            timeout=20
```
Comment (FrancoLionti) [inline src/python_testing/TC_SU_2_7.py:140]: Just asking, but what was the motivation for this increase? Im guessing that all added time guards you against the possibility of a pesky timeout but just confirming so. If this is the case wouldnt the actual network speed be the biggest concern ?

File: src/python_testing/TC_SU_2_7.py [hunk 2]
```diff
@@ -320,7 +228,7 @@ async def test_TC_SU_2_7(self):
         logger.info("About close the provider app with proc %s", self.current_provider_app_proc)
         self.terminate_provider()
```
Comment (FrancoLionti) [inline src/python_testing/TC_SU_2_7.py:228]: Im guessing that the previous timer increase was because this step was failing but just some guesswork. Im thinking that device should reboot here as it is done in other steps. Maybe this collides with the new definition of the first step but idk.
```suggestion
        self.terminate_provider()
        await self.request_device_reboot()

```
Im saying this because in DefaultOTARequestorDriver.cpp there's this condition:` if (mRequestor->GetCurrentUpdateState() != OTAUpdateStateEnum::kIdle)`

And it will ignore the announcement.

**Review by FrancoLionti — APPROVED** [review]
There's one minor comment overall. Also i think the title is wrong since it says 2.5 and the file is 2.7. 

If possible i would also write down in text the steps right now its somewhat hard to read the pic. (No im not blind just mildly vissualy impaired 😆)

File: src/python_testing/TC_SU_2_7.py (hunk 1 above)
Comment (jtrejoespinoza-grid) [reply src/python_testing/TC_SU_2_7.py:140]: I remember in the past we had issues sometimes with the app starting, but most of those cases where caused by other conditions we did not have into consideration. I will update all to 20 seconds to keep it standard.

File: src/python_testing/TC_SU_2_7.py (hunk 2 above)
Comment (jtrejoespinoza-grid) [reply src/python_testing/TC_SU_2_7.py:228]: Here reboot is not needed because the other states are not validated, we just care if we get into the State of kFailure after the send Announce Ota provider without a provider available.

File: src/python_testing/TC_SU_2_7.py
Comment (FrancoLionti) [inline src/python_testing/TC_SU_2_7.py:32]: Just read Andrei and Cecille's comment on slack, it seems this introduced a regression in the nightly TC_SU_2_7 run. It's been failing since this merged. As far as i can tell --app-pipe/--app-pipe-out were added to script-args but not mirrored in app-args for ${OTA_REQUESTOR_APP}. The test script now requires the named pipe at startup as well (runner.py::convert_args_to_matter_config), but since the requestor app is never launched with --app-pipe, i think it never makes the FIFO, so the test fails immediately with:
FileNotFoundError: CANNOT FIND '/tmp/requestor_2_7_fifo' before it even reaches step 1. I think it was not possible to check it in the current PR since this regression is on a slow test workflow.

Just cheked the logs and a possible fix could be to add the following:

```suggestion
#       --app-pipe /tmp/requestor_2_7_fifo
#       --app-pipe-out /tmp/requestor_2_7_fifo_out
```

Hope this helps!

Comment (jtrejoespinoza-grid) [reply src/python_testing/TC_SU_2_7.py:32]: In this case is the opposite, the issue is that this argument was kept by mistake and its not needed anymore. In this case I think I removed from the app args but not from the script arguments when testing manually causing the issue.

**Review by cecille — APPROVED** [review]

**Review by cecille — APPROVED** [review]

**Comment by cecille** [comment]
This has now been sitting for two weeks with no further reviews. Labeling.
